src/screens/repo/screens/build/menu.js

```diff
@@ -115,25 +116,25 @@ export default class BuildMenu extends Component {
 								)}
 							</li>
 							<li>
-								{build.status === "success" ? (
+								{build.status === STATUS_SUCCESS  ? (
 									<button onClick={this.togglePromote}>
 										<DeployIcon />
 										<span>Promote Build</span>
 									</button>
 								):null }
+                                {build.status === STATUS_SUCCESS  && this.state.togglePromote ? (
+                                    <ul class="sub">{envs.map(function(env) {
+                                        return <li><button onClick={handlePromote.bind(this, env)}><PlayIcon /><span>{env}</span></button></li>
+                                    })}
+                                    <li><button onClick={handlePromote.bind(this, this.state.customEnv)}><PlayIcon/>
+                                        <input type="text" value={this.state.customEnv}
+                                               onClick={event => event.stopPropagation()}
+                                               onChange={this.updateCustomEnv.bind(this)}
+                                               placeholder="input and then click play icon">
```

Comment:
Placeholder description should indicate the content of field and is not intended to be a explanation on how to use it, text should be something like: `Deployment target (eg: staging)`